### DISCUSSION TITLE
Clarifications on using JWT session strategy despite using an adapter

### DISCUSSION THREAD

**Discussion by ChristianIvicevic** [nextauthjs/next-auth#4980] [Help]
During my experiments I noticed the `withAuth` middleware that I got working only after setting the session strategy to JWT. Of course I was curious about the implications and read about the details in the FAQ over here: https://next-auth.js.org/faq#json-web-tokens

The documentation however seems to be contradiccting itself. In the linked FAQ it claims the following
> What are the disadvantages of JSON Web Tokens?
>
> [...] NextAuth.js client includes advanced features to mitigate the downsides of using shorter session expiry times on the user experience, including automatic session token rotation [...]

However it also claims the following

> How do I get Refresh Tokens and Access Tokens for an OAuth account?
>
> [...] Note: NextAuth.js does not currently handle Access Token rotation for OAuth providers for you [...]

Now I am curious whether falling back to JWT implies that I have to implement said rotation based on the example code or not.

Furthermore, if I decide I want to stick to sessions persisted in the database without JWT what's the approach to handle auth without using the middleware? Just a naive `useSession` on the respective pages and redirecting to login if it's undefined?

**Comment by rbluethl** [comment]
Hey @ChristianIvicevic! Wondering about the exact same thing. I read the documentation back and forth and don't quite understand what happens when setting `strategy: jwt` while using an adapter. Furthermore, as you said, the `middleware.ts` handling doesn't seem to work when using an adapter. Curious if there's anything you found out since posting this!

**Reply by ChristianIvicevic** [reply]
So the token refresh is done automatically, there is no need to do anything, when using JWT. If you want to have session-based auth in the database you cannot use the middleware and have to do that in gSSP for which you will probably write a higher-order function to reduce boilerplate.
